docs/Dev_eXo_Addons.rst

```diff
@@ -1139,11 +1150,173 @@ following:
+          return result;
+        }
+      }
+
+* A :ref:`Portal extension <PLFDevGuide.eXoAdd-ons.PortalExtension>` configuration file:
+
+   .. code:: xml
+
+      <external-component-plugins>
+        <target-component>org.exoplatform.addon.wallet.reward.service.RewardSettingsService</target-component>
+        <component-plugin>
+          <!-- Reward plugin name: must be unique  -->
+          <name>custom</name>
+          <set-method>registerPlugin</set-method>
+          <!-- FQN of plugin class  -->
+          <type>org.example.CustomRewardPlugin</type>
+          <description>Custom reward plugin</description>
+        </component-plugin>
+      </external-component-plugins>
+
+One you deployed your plugin, you will see it added in Reward administration UI:
+
+|CustomRewardPlugin|
+
+.. _PLFDevGuide.eXoAdd-ons.WalletListeners:
+
+==============================
+eXo Wallet listeners
+==============================
+
+eXo Wallet uses :ref:`ListenerService <Kernel.UnderstandingtheListenerService>` to broadcast events about wallets and transactions lifecycle.
+
+Broadcasted events are:
+
+- ``exo.addon.wallet.addressAssociation.new`` : a new wallet gets created for the first time by a user. (Example: `NewWalletListener <https://github.com/exoplatform/wallet/blob/develop/wallet-services/src/main/java/org/exoplatform/addon/wallet/listener/NewWalletListener.java>`__)
+- ``exo.addon.wallet.addressAssociation.modification`` : a user/space's associated wallet address is modified. (Example: `ModifiedWalletListener <https://github.com/exoplatform/wallet/blob/develop/wallet-services/src/main/java/org/exoplatform/addon/wallet/listener/ModifiedWalletListener.java>`__)
+- ``exo.addon.wallet.transaction.mined`` : a pending transaction sent from a knwon wallet address gets mined on the blockchain and updated in internal database. (Example: `TransactionNotificationListener <https://github.com/exoplatform/wallet/blob/develop/wallet-services/src/main/java/org/exoplatform/addon/wallet/listener/TransactionNotificationListener.java>`__)
+
+To add an event listener using one listed events above, you can add the following configuration inside a  :ref:`Portal extension <PLFDevGuide.eXoAdd-ons.PortalExtension>` configuration file:
+
+   .. code:: xml
+
+      <external-component-plugins>
+        <target-component>org.exoplatform.services.listener.ListenerService</target-component>
+        <component-plugin>
+          <!-- One of the listed event names -->
+          <name>EVENT_NAME</name>
+          <set-method>addListener</set-method>
+          <!-- FQN of the event listener -->
+          <type>org.example.CustomEventListener</type>
+        </component-plugin>
+      </external-component-plugins>
+
+.. _PLFDevGuide.eXoAdd-ons.KudosListeners:
+
+==============================
+eXo Kudos listeners
+==============================
+
+eXo Kudos uses :ref:`ListenerService <Kernel.UnderstandingtheListenerService>` to broadcast events about kudos lifecycle.
```

Comment:
```suggestion
eXo Kudos uses :ref:`ListenerService <Kernel.UnderstandingtheListenerService>` to broadcast events about kudos lifecycle.
```
Developers can leverage these events to build custom features or alter external systems. Internally, these events are used for a variety of uses such as notifications or maintaining wallet and transactions statuses.

Reply:
Done